src/nemo_evaluator/observability/scrape_metrics.py

```
@@ -91,26 +93,52 @@ async def _poll_once(session: aiohttp.ClientSession, url: str) -> dict[str, Any]
-                record = await _poll_once(session, url)
+                record = await _poll_once(session, url, labels=labels)
                 fp.write(json.dumps(record, separators=(",", ":")) + "\n")
                 fp.flush()
                 with contextlib.suppress(asyncio.TimeoutError):
                     await asyncio.wait_for(stop_event.wait(), timeout=interval_s)
     return 0
 
 
+def _parse_label(spec: str) -> tuple[str, str]:
+    if "=" not in spec:
+        raise argparse.ArgumentTypeError(f"--label requires NAME=VALUE, got {spec!r}")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)